tsidp-server.go

```diff
@@ -186,6 +187,34 @@ func main() {
 
 	slog.Info("tsidp server started", slog.String("server_url", srv.ServerURL()))
 
+	if *flagUnixSocket != "" {
+		socketPath := *flagUnixSocket
+		info, err := os.Stat(socketPath)
+		if err == nil && (info.Mode()&os.ModeSocket) != 0 {
+			// A socket file already exists.
+			c, err := net.Dial("unix", socketPath)
+			if err == nil {
+				c.Close()
+				slog.Error("unix socket already in use")
+				os.Exit(1)
+			}
+
+			// It's a stale socket, so we can remove it.
+			os.Remove(socketPath)
+		}
+
+		ln, err := net.Listen("unix", *flagUnixSocket)
+		if err != nil {
+			slog.Error("failed to listen on unix socket", slog.Any("error", err))
+			os.Exit(1)
+		}
+		defer func() {
+			ln.Close() // TODO: the other listeners are not closed?
```

Comment:
I also found this somewhat strange, I'm fairly sure that unix sockets _must_ be closed, but I noticed that none of the tcp listeners are being closed. Are they cleaned up automatically by go when the program exits?